File: classifier/data/raw/ai/samples/project_euler_java/Problem069.java
// Project Euler 69: n <= one million maximizing n / phi(n).

public class Problem069 {
    public static void main(String[] args) {
        int limit = 1_000_000;
        int product = 1;
        for (int candidate = 2; ; candidate++) {
            if (!prime(candidate)) continue;
            if (product > limit / candidate) break;
            product *= candidate;
        }
        System.out.println(product);
    }

    static boolean prime(int n) {
        for (int d = 2; d * d <= n; d++) if (n % d == 0) return false;
        return true;
    }
}

File: classifier/data/raw/ai/samples/project_euler_zig/problem_069.zig
const std = @import("std");

// Project Euler 69: n <= one million maximizing n / phi(n).
fn isPrime(n: u64) bool {
    if (n < 2) return false;
    var divisor: u64 = 2;
    while (divisor * divisor <= n) : (divisor += 1) {
        if (n % divisor == 0) return false;
    }
    return true;
}

pub fn main() !void {
    const limit: u64 = 1_000_000;
    var product: u64 = 1;
    var candidate: u64 = 2;
    while (true) : (candidate += 1) {
        if (!isPrime(candidate)) continue;
        if (product * candidate > limit) break;
        product *= candidate;
    }
    var buffer: [32]u8 = undefined;
    const text = try std.fmt.bufPrint(&buffer, "{d}\n", .{product});
    _ = std.c.write(1, text.ptr, text.len);
}
